cobalt/watchdog/watchdog.cc

```diff
@@ -579,23 +579,44 @@ std::string Watchdog::GetWatchdogViolations(
             violations_count_ -= violations_count;
             if (!static_cast<base::DictionaryValue*>(violations_map_.get())
                      ->empty()) {
-              WriteWatchdogViolations();
+              pending_write_ = true;
             } else {
               starboard::SbFileDeleteRecursive(GetWatchdogFilePath().c_str(),
                                                true);
             }
           }
         }
       }
-      if (!filtered_client_data.DictEmpty()) {
-        base::JSONWriter::Write(filtered_client_data, &watchdog_json_fetched);
+      if (!fetched_violations.DictEmpty()) {
+        base::JSONWriter::Write(fetched_violations, &fetched_violations_json);
       }
+      EvictOldWatchdogViolations();
     }
-    SB_LOG(INFO) << "[Watchdog] Reading violations:\n" << watchdog_json_fetched;
+    SB_LOG(INFO) << "[Watchdog] Reading violations:\n"
+                 << fetched_violations_json;
   } else {
     SB_LOG(INFO) << "[Watchdog] No violations.";
   }
-  return watchdog_json_fetched;
+  return fetched_violations_json;
+}
+
+void Watchdog::EvictOldWatchdogViolations() {
```

Comment:
We should add a test for this function as well.

Reply:
added a test. also significantly trimmed GetPartialViolationsByClients test. this was because that test was failing due to testing beyond its scope. it was testing for file writes but that behavior has now changed from always forced to next available monitor loop or shutdown (those 2 write cases are already being tested by other more specific tests)